MolecularNodes/load.py

```diff
@@ -374,3 +374,39 @@ def att_is_carb():
         warnings.warn('No chain information detected.')
     
     return mol_object, coll_frames
+
+
+def load_star_file(file_path, obj_name = 'Star Instances'):
+    import starfile
+    from scipy.spatial.transform import Rotation as R
+    
+    star = starfile.read(file_path)
+
+    df = star['particles'].merge(star['optics'], on='rlnOpticsGroup')
+
+    # get necessary info from dataframes
+    xyz = df[['rlnCoordinateX', 'rlnCoordinateY', 'rlnCoordinateZ']].to_numpy()
+    shifts_ang = df[['rlnOriginXAngst', 'rlnOriginYAngst', 'rlnOriginZAngst']].to_numpy()
+    pixel_size = df['rlnImagePixelSize'].to_numpy()
+    euler_angles = df[['rlnAngleRot', 'rlnAngleTilt', 'rlnAnglePsi']].to_numpy()
+
+    # Get absolute position and orientations
+    # shifting currently doesn't work - get errors with broadcasting
+    # particle_positions = xyz - (shifts_ang / pixel_size)
+    
+    # currently doubling up on euler transformations, needs cleaning up
+    rotation_matrices = R.from_euler(
+        seq='ZYZ', angles=euler_angles, degrees=True
+    ).inv().as_matrix()
+
+    eulers = R.from_matrix(rotation_matrices).as_euler('xyz')
+
+    obj = create_object(obj_name, coll_mn(), xyz / 1e3)
```

Comment:
Currently I am scaling everything inside of molecular nodes by 0.01, which for the `.pdb` files etc makes 1 Å = 1 cm. 

I'm not _super_ happy with this scaling, as it doesn't match nicely with any powers of 10, but in terms of a practicality standpoint it works out best for new users jumping into Blender. When they import a protein this way, they see it in front of the camera as they would expect, and not 10 times smaller or 100 times larger. In the future there is potential for more 'dedicated' builds for Blender at which point I might change the world scale to 'work' better. Anyway, I digress.

In this code, had just scaled it down so I could actually see the model to test things out, and this won't be what the final code should look like. What are the units inside of relion files? nm or Å? In my other functions I have a `world_scale` value that I scale all of the positions by so I suppose we should do something similar here to allow for changes in the future.

Reply:
Gotcha - that scaling makes total sense, (I guess the perspective matrices in blender are setup to work well for 'real world' distances?)

The coordinates in the file are provided in pixel coordinates within a 3D reconstruction so are a bit arbitrary - normalising to angstroms probably makes the most sense as a first step, multiplying the coordinates by the pixel size column instead of dividing the shifts